### PR TITLE
get things to compile on jdk11

### PR DESCRIPTION
**Pull request by xuwei-k**

### CODE REVIEW

**Comment by typesafe-tools** [comment]
To validate this pull request against other sbt modules, please visit <A HREF="https://jenkins.scala-sbt.org:8592/job/sbt-validator/parambuild/?sbt=develop&zinc=pull/602/head&io=develop&librarymanagement=develop&util=develop&website=develop&deploy_to_bintray=false&run_sbt_community=false&verbose_dbuild=false&community_verbose=false">this link.</A>

File: internal/zinc-persist/src/main/scala/sbt/internal/inc/text/Base64.scala [hunk 1]
```diff
@@ -30,10 +29,28 @@ private[sbt] object Base64 {
...
 
+private[sbt] object Java678Encoder {
+  private[this] val DatatypeConverter_class = Class.forName("javax.xml.bind.DatatypeConverter")
+  private val DatatypeConverter_encode =
+    DatatypeConverter_class.getMethod("printBase64Binary", classOf[Array[Byte]])
+  private val DatatypeConverter_decode =
+    DatatypeConverter_class.getMethod("parseBase64Binary", classOf[String])
+}
+
 private[sbt] class Java678Encoder extends Base64 {
```
Comment (xuwei-k) [inline internal/zinc-persist/src/main/scala/sbt/internal/inc/text/Base64.scala]: BTW, Is there any reasons `Java678Encoder` exists? zinc still support Java 7? 🤔 

https://github.com/sbt/zinc/issues/451#issuecomment-342408977

>  we can target anything less than Java 8 given we depend on JDK 8 classes

Comment (eed3si9n) [reply internal/zinc-persist/src/main/scala/sbt/internal/inc/text/Base64.scala]: We can drop Java 7 workarounds.

Comment (xuwei-k) [reply internal/zinc-persist/src/main/scala/sbt/internal/inc/text/Base64.scala]: removed

**Review by jvican — COMMENTED** [review]
Thank you @xuwei-k. I've added two commits, one to test JDK 11 in the Drone CI and the other one to fix a MIMA error. However, we still get some errors that seem legit:
  
```
[error] /drone/src/github.com/sbt/zinc/internal/zinc-apiinfo/src/main/scala/xsbt/api/HashAPI.scala:165:46: value name is not a member of Package
119s
318
[error]   def hashPackage(p: Package) = hashString(p.name)
119s
319
[error]                                              ^
119s
320
[warn] /drone/src/github.com/sbt/zinc/internal/zinc-apiinfo/src/main/scala/xsbt/api/HashAPI.scala:165:19: parameter value p in method hashPackage is never used
119s
321
[warn]   def hashPackage(p: Package) = hashString(p.name)
119s
322
[warn]                   ^
119s
323
[error] /drone/src/github.com/sbt/zinc/internal/zinc-apiinfo/src/main/scala/xsbt/api/Visit.scala:22:19: value name is not a member of Package
120s
324
[error]     visitString(p.name)
120s
325
[error]                   ^
```
  
They happen when MiMa runs too but they look related to a change in the JDK APIs.

**Comment by jvican** [comment]
Wait, it looks like the last commit passed! I'm going to add another commit to make sure the JAVA_HOME is the proper one, and if it is then I merge.

**Comment by jvican** [comment]
Alrighty, now we're talking. It looks like we still need a little bit more work to get JDK 11 working. @xuwei-k If you don't have time to do this I would be happy to take over some time next week.
  
Here's the log describing some of the problems: https://ci.scala-lang.org/sbt/zinc/99/8

**Comment by jvican** [comment]
Unfortunately, I cannot prioritize this, so I won't have time to look at it next week as I said in my previous comment

**Comment by jvican** [comment]
See comment in https://github.com/scalacenter/bloop/issues/588#issuecomment-463135020

**Comment by eed3si9n** [comment]
Closing this in favor of #640

**Pull request closed without merging**